app/src/main/java/com/jetbrains/handson/mpp/mobile/TrainDataAdapter.kt

```diff
@@ -78,10 +78,26 @@ class TrainDataAdapter(private val dataSet: List<TrainTimes.Journey>) :
     override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
         // - get element from your dataset at this position
         // - replace the contents of the view with that element
-        if (position == 0) {
+        if (position == 0) { // the first position will be the heading
             holder.addTitle()
         } else {
             holder.addJourney(dataSet[position-1])
+            holder.view.setOnClickListener { view ->
+                val intent = Intent(view.context, JourneyDetailActivity::class.java)
+                intent.putExtra("status", holder.journey?.status)
+                intent.putExtra("primaryOperator", holder.journey?.trainOperator)
+                intent.putExtra("stationChangeNames", holder.journey?.changes?.map {
+                    it.name
+                }?.toTypedArray())
+                val diff = holder.journey?.arrivalTime!! - holder.journey?.departureTime!!
+                val hours = TimeUnit.MILLISECONDS.toHours(diff)
+                val min = TimeUnit.MILLISECONDS.toMinutes(diff) - hours*60
+                intent.putExtra("duration", if (hours == 0L)
+                    "$min minutes"
+                else
+                    "$hours hours, $min minutes")
```

Comment:
Do we need a `diffFormatted` in `Journey`?

Reply:
Not for now.